src/server/pkg/storage/fileset/util.go

```diff
@@ -12,3 +17,43 @@ func WithLocalStorage(f func(*Storage) error) error {
 		return f(NewStorage(objC, chunks))
 	})
 }
+
+func CopyFiles(w *Writer, r FileSource) error {
+	switch r := r.(type) {
+	case *Reader:
+		return r.iterate(func(fr *FileReader) error {
+			return w.CopyFile(fr)
+		})
+	default:
+		return errors.Errorf("CopyFiles does not support reader type: %T", r)
+	}
+}
+
+// WriteTarEntry writes an tar entry for f to w
+func WriteTarEntry(w io.Writer, f File) error {
+	h, err := f.Header()
+	if err != nil {
+		return err
+	}
+	tw := tar.NewWriter(w)
+	if err := tw.WriteHeader(h); err != nil {
+		return err
+	}
+	if (h.Mode == tar.TypeReg || h.Mode == tar.TypeRegA) && h.Size > 0 {
```

Comment:
You shouldn't need this check. `Content` should write zero bytes when the size is 0.